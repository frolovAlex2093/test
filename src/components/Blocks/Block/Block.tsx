import { Box, Checkbox, FormControlLabel, Paper, Typography } from '@mui/material';
import React from 'react';
import { IBlock } from '../../../interfaces/interfaces';
import { BlockItem } from './BlockItem/BlockItem';

export const Block: React.FC<IBlock> = ({ blocksItem, onChangeBlock }) => {
  return (
    <>
      {blocksItem.map((item, index) => {
        return (
          <Box key={index}>
            <Paper
              elevation={10}
              sx={{
                marginBottom: '15px',
                padding: ' 15px 20px',
                borderRadius: 1,
                gap: 2,
                border: 'solid #bdbdbd 1px'
              }}
            >
              <Typography variant='h6' component='h6' gutterBottom color='black'>
                {item.blockName}
              </Typography>

              {item.checkbox ? (
                item.check ? (
                  <FormControlLabel
                    control={
                      <Checkbox
                        onChange={() => onChangeBlock(item.id)}
                        checked={item.check}
                      ></Checkbox>
                    }
                    label='Отсутствует'
                  />
                ) : (
                  <>
                    <FormControlLabel
                      control={
                        <Checkbox
                          onChange={() => onChangeBlock(item.id)}
                          checked={item.check}
                        ></Checkbox>
                      }
                      label='Отсутствует'
                    />
                    <BlockItem blockItem={item.blockItem}></BlockItem>
                  </>
                )
              ) : (
                <BlockItem blockItem={item.blockItem}></BlockItem>
              )}
            </Paper>
          </Box>
        );
      })}
    </>
  );
};
