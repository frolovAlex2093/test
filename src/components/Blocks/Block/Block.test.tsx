import { MemoryRouter } from 'react-router-dom';

import { render, screen } from '@testing-library/react';
import { Block } from './Block';
import userEvent from '@testing-library/user-event';

const onChangeBlock = jest.fn();

const blockItem = [
  {
    blockName: 'test name 1',
    id: 40,
    blockItem: [
      {
        name: 'test name 1_1',
        type: 'TextField',
        id: 145,
        require: false,
        checkbox: false,
        disabled: false,
        value: ['']
      }
    ]
  },
  {
    blockName: 'test name 2',
    id: 40,
    checkbox: true,
    check: false,
    blockItem: [
      {
        name: 'test name 2_1',
        type: 'TextField',
        id: 145,
        require: false,
        checkbox: false,
        disabled: false,
        value: ['']
      }
    ]
  }
];

describe('Block component', () => {
  it('Block component render', () => {
    render(<Block blocksItem={blockItem} onChangeBlock={onChangeBlock}></Block>);
    expect(screen.getByText('test name 1')).toBeInTheDocument();
    expect(screen.getByText('test name 2')).toBeInTheDocument();

    expect(screen.getByLabelText('test name 1_1')).toBeInTheDocument();
    expect(screen.getByLabelText('test name 2_1')).toBeInTheDocument();

    expect(screen.getByText('Отсутствует')).toBeInTheDocument();
  });
  it('Block component function', () => {
    render(<Block blocksItem={blockItem} onChangeBlock={onChangeBlock}></Block>);

    userEvent.click(screen.getByText('Отсутствует'));

    expect(onChangeBlock).toHaveBeenCalled();
  });
  it('Block component snapshot', () => {
    const blockSnap = render(<Block blocksItem={blockItem} onChangeBlock={onChangeBlock}></Block>);
    expect(blockSnap).toMatchSnapshot();
  });
});
