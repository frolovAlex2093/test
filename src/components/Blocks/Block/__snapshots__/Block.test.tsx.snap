// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block component Block component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-1g1fz7u-MuiPaper-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1u7f1pc-MuiTypography-root"
          >
            test name 1
          </h6>
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <div
                class="MuiBox-root css-16a3qdz"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-156lfgm-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="145"
                    id="145-label"
                  >
                    test name 1_1
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                      id="145"
                      pattern=".{0,}"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                      />
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          test name 1_1
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-1g1fz7u-MuiPaper-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1u7f1pc-MuiTypography-root"
          >
            test name 2
          </h6>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Отсутствует
            </span>
          </label>
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <div
                class="MuiBox-root css-16a3qdz"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-156lfgm-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="145"
                    id="145-label"
                  >
                    test name 2_1
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                      id="145"
                      pattern=".{0,}"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                      />
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          test name 2_1
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-1g1fz7u-MuiPaper-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1u7f1pc-MuiTypography-root"
        >
          test name 1
        </h6>
        <div
          class="MuiBox-root css-j7qwjs"
        >
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiBox-root css-16a3qdz"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-156lfgm-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="145"
                  id="145-label"
                >
                  test name 1_1
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id="145"
                    pattern=".{0,}"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                    />
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        test name 1_1
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-1g1fz7u-MuiPaper-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1u7f1pc-MuiTypography-root"
        >
          test name 2
        </h6>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Отсутствует
          </span>
        </label>
        <div
          class="MuiBox-root css-j7qwjs"
        >
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiBox-root css-16a3qdz"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-156lfgm-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="145"
                  id="145-label"
                >
                  test name 2_1
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id="145"
                    pattern=".{0,}"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                    />
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        test name 2_1
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
